rm/rm-server/src/test/java/functionaltests/utils/RMTHelper.java

```diff
@@ -138,6 +138,10 @@ public void createNodeSource(String name, int nodeNumber) throws Exception {
         createNodeSource(name, nodeNumber, getResourceManager(), getMonitorsHandler());
     }
 
+    public void createNodeSourceWithInfiniteTimeout(String name, int nodeNumber) throws Exception {
```

Comment:
I think that wrapping the exceptions into RuntimeException will avoid the exceptions propagation and will improve the code readability.  This comment should be applied to the following methods too.

Reply:
I think you ask for refactoring which is not related to this PR.

Reply:
I think that refactoring when adding a new functionality can  be part of the pull request. It is better to improve the code in each pull request than doing a huge refactoring when the code is not working anymore because each developer just keep adding new features. Moreover adding new functionality often highlight design issues.